core/src/main/java/com/graphhopper/routing/util/parsers/BikeCommonAccessParser.java

```diff
@@ -42,61 +37,87 @@ protected BikeCommonAccessParser(BooleanEncodedValue accessEnc, BooleanEncodedVa
     public WayAccess getAccess(ReaderWay way) {
         String highwayValue = way.getTag("highway");
         if (highwayValue == null) {
-            WayAccess access = WayAccess.CAN_SKIP;
+            return checkNonHighwayAccess(way);
+        }
 
-            if (FerrySpeedCalculator.isFerry(way)) {
-                // if bike is NOT explicitly tagged allow bike but only if foot is not specified either
-                String bikeTag = way.getTag("bicycle");
-                if (bikeTag == null && !way.hasTag("foot") || intendedValues.contains(bikeTag))
-                    access = WayAccess.FERRY;
-            }
+        if (!allowedHighways.contains(highwayValue))
+            return WayAccess.CAN_SKIP;
+
+        String bicycleValue = way.getTag("bicycle");
+        boolean hasAllowedBikesExplicitly = intendedValues.contains(bicycleValue);
```

Comment:
Nit: I'd say, only calculate what you need. In line 50, there is a return  that does not depend on line 47, so this calculation should be moved below the if below this line IMHO.